### PR TITLE
root: replace builtin psycopg libpq binary implementation with distro…

### PR DESCRIPTION
**Pull request by rissson**
… provided one

## Details

As discussed previously.

https://www.psycopg.org/psycopg3/docs/basic/install.html#local-installation recommends using the C implementation when possible.

REPLACE ME

---

## Checklist

-   [ ] Local tests pass (`ak test authentik/`)
-   [ ] The code has been formatted (`make lint-fix`)

If an API change has been made

-   [ ] The API schema has been updated (`make gen-build`)

If changes to the frontend have been made

-   [ ] The code has been formatted (`make web`)
-   [ ] The translation files have been updated (`make i18n-extract`)

If applicable

-   [ ] The documentation has been updated
-   [ ] The documentation has been formatted (`make website`)

### CODE REVIEW

**Comment by rissson** [comment]
For reference and comparison:
CI on main: https://github.com/goauthentik/authentik/actions/runs/5730041231/job/15527967561
CI on branch: https://github.com/goauthentik/authentik/actions/runs/5731583764/job/15532788971?pr=6448

**Comment by rissson** [comment]
Setting up the dev env in the CI takes about the same time (2-3 minutes)

File: Dockerfile [hunk 1]
```diff
@@ -81,13 +81,13 @@ COPY --from=geoip /usr/share/GeoIP /geoip
...
-    apt-get install -y --no-install-recommends build-essential pkg-config libxmlsec1-dev zlib1g-dev && \
+    apt-get install -y --no-install-recommends build-essential pkg-config libxmlsec1-dev zlib1g-dev libpq-dev python3-dev && \
     # Required for runtime
-    apt-get install -y --no-install-recommends libxmlsec1-openssl libmaxminddb0 && \
+    apt-get install -y --no-install-recommends libpq5 openssl libxmlsec1-openssl libmaxminddb0 && \
     # Required for bootstrap & healtcheck
     apt-get install -y --no-install-recommends runit && \
     pip install --no-cache-dir -r /requirements.txt && \
-    apt-get remove --purge -y build-essential pkg-config libxmlsec1-dev && \
+    apt-get remove --purge -y build-essential pkg-config libxmlsec1-dev libpq-dev python3-dev && \
```
[the remark this answers is not in the document]
Comment (rissson) [reply Dockerfile:90]: After checking what `python3-dev` and dependencies install vs what's already bundled in the image, I don't think it will.

**Comment by rissson** [comment]
Docker image size is 1MB heavier with this PR.

**Comment by BeryJu** [comment]
> Docker image size is 1MB heavier with this PR.

I was just about to comment on that, they look the same size here

```
ghcr.io/goauthentik/dev-server          gh-psycopg-c-1690924082-70dfd86-arm64                   cb2f6d67160f   6 minutes ago   798MB
ghcr.io/goauthentik/dev-server          gh-next-1690912046-cc6824f-arm64                        db264d6c234a   3 hours ago     798MB
```

**Review by BeryJu — APPROVED** [review]

**Comment by rissson** [comment]
I just realized but this may break some development environments if those packages are not installed. I'll put in a follow up PR to update the dev docs

**Comment by BeryJu** [comment]
> I just realized but this may break some development environments if those packages are not installed. I'll put in a follow up PR to update the dev docs

I did wonder about that too, but it installed just fine for me locally without any changes

We do already tell people to install `libpq` so it should all be fine?

**Comment by rissson** [comment]
Yep, it will be fine indeed

**Pull request merged**